subsys/bluetooth/audio/bap_broadcast_source.c

```diff
@@ -885,6 +961,24 @@ int bt_bap_broadcast_source_reconfig(struct bt_bap_broadcast_source *source,
 
 			stream_param = &subgroup_param->params[j];
 			stream = stream_param->stream;
+			if (CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0) {
+				if (bis_count >= BROADCAST_STREAM_CNT) {
+					LOG_DBG("Stream count %d exceeded", bis_count);
+					return -ENOMEM;
+				}
+				codec_cfg = &source->codec_cfg[bis_count];
+				memcpy(codec_cfg, subgroup_param->codec_cfg,
+				       sizeof(struct bt_audio_codec_cfg));
+				err = update_codec_cfg_data(codec_cfg, stream_param);
+
+				if (err != 0) {
+					LOG_DBG("codec config update failed [%zu]: %d", i, err);
+					broadcast_source_cleanup(source);
+					return err;
```

Comment:
```suggestion
					LOG_DBG("codec config update failed [%zu]: %d", i, err);
					return err;
```

We should not cleanup the broadcast source if reconfiguration fails